src/braket/circuits/serialization.py

```diff
@@ -38,15 +38,20 @@ class SerializableProgram(ABC):
     def to_ir(
         self,
         ir_type: IRType = IRType.OPENQASM,
+        allow_implicit_build: bool = True,
```

Comment:
As mentioned in the PR description, I'll make a follow-up PR which flips this default to `False` and makes the resulting avalanche of downstream changes, since every existing call to `.to_ir()` will need to be updated to `.build().to_ir()`.